docs/hugo/content/contributing/extension-points/deleter.md

```diff
@@ -0,0 +1,345 @@
+    }
+
+    // Step 2: Wait for dependent resources to be cleaned up
+    if ex.hasDependents(ctx, resource) {
+        log.V(Status).Info("Waiting for dependent resources to be deleted")
+        return ctrl.Result{RequeueAfter: 10 * time.Second}, nil
+    }
+
+    // Step 3: Proceed with deletion
+    log.V(Status).Info("All prerequisites met, proceeding with deletion")
+    return next(ctx, log, resolver, armClient, obj)
+}
+```
+
+### Pattern 4: Soft Delete with Purge Option
+
+```go
+func (ex *ResourceExtension) Delete(
+    ctx context.Context,
+    log logr.Logger,
+    resolver *resolver.Resolver,
+    armClient *genericarmclient.GenericClient,
+    obj genruntime.ARMMetaObject,
+    next extensions.DeleteFunc,
+) (ctrl.Result, error) {
+    resource := obj.(*myservice.MyResource)
+
+    // Perform standard deletion (moves to soft-deleted state)
+    result, err := next(ctx, log, resolver, armClient, obj)
+    if err != nil {
+        return result, err
+    }
+
+    // If purge is requested, purge the soft-deleted resource
+    if resource.Spec.DeleteMode != nil && *resource.Spec.DeleteMode == "Purge" {
+        log.V(Status).Info("Purging soft-deleted resource")
+        if err := ex.purgeResource(ctx, resource, armClient); err != nil {
+            return ctrl.Result{}, eris.Wrap(err, "failed to purge resource")
+        }
+    }
+
+    return ctrl.Result{}, nil
+}
+```
+
+## Deletion Lifecycle
+
+Understanding the deletion process:
+
+1. **User deletes resource**: `kubectl delete` sets deletion timestamp
+2. **Finalizer blocks deletion**: ASO finalizer prevents immediate removal from Kubernetes
+3. **Deleter invoked**: Custom `Delete()` method is called
+4. **Pre-deletion logic**: Extension performs custom operations
+5. **Standard deletion**: `next()` sends DELETE to ARM
+6. **ARM deletion completes**: Azure resource is removed
+7. **Finalizer removed**: Kubernetes removes the resource
+
+If any step fails, the process pauses and will retry on the next reconciliation.
+
+## Error Handling
+
+Proper error handling in deleters is critical:
+
+```go
+// Transient error - will retry
+return ctrl.Result{}, eris.Wrap(err, "temporary failure")
+
+// Permanent error with condition
+return ctrl.Result{}, conditions.NewReadyConditionImpactingError(
+    err,
+    conditions.ConditionSeverityError,
+    conditions.ReasonFailed)
+
+// Requeue for later retry
+return ctrl.Result{RequeueAfter: 1 * time.Minute}, nil
+
+// Success
+return ctrl.Result{}, nil
+```
+
+## Testing
+
+When testing `Deleter` extensions:
+
+1. **Test successful deletion**: Verify the happy path works
+2. **Test pre-deletion operations**: Ensure cleanup logic executes
+3. **Test error scenarios**: Verify error handling prevents finalizer removal
+4. **Test idempotency**: Multiple calls should be safe
+5. **Test conditional paths**: Cover all branching logic
+6. **Test requeue behavior**: Verify multi-step deletions requeue correctly
+
+Example test structure:
```

Comment:
Not sure we need to see an example test structure here? Seems a bit overkill for the docs.